src/GridFS/ReadableStream.php

```diff
@@ -186,9 +186,26 @@ public function seek($offset)
         $this->chunkOffset = (integer) floor($offset / $this->chunkSize);
         $this->bufferOffset = $offset % $this->chunkSize;
 
-        if ($lastChunkOffset !== $this->chunkOffset) {
+        /* If we are seeking to a previous chunk, we need to reinitialize the
```

Comment:
An idea for an early guard statement might be:

```php
if ($lastChunkOffset === $this->chunkOffset) {
    return;
}

// Clear the buffer as the current chunk will be changed
$this->buffer = null;

// If we are seeking to a previous chunk...
```

If the chunk isn't changing at all, we have nothing more to do after updating `$chunkOffset` and `$bufferOffset` a few lines up. Otherwise, we know the chunk will change, so the existing `$buffer` is no longer relevant and should be cleared (`readBytes()` will reinitialize it when needed). That should address the bug I mentioned below and in the related test PR.

From there, I think your existing logic is correct:

 * If we're seeking to a previous chunk, reset `$chunksIterator` and return (`readBytes()` will reinitialize it when needed)
 * If we're seeking to a later chunk, advance until we reach that chunk

This should also make it easier to add logic that prefers resetting `$chunksIterator` if we're seeking _very_ far ahead in the file stream. This is something I alluded to in the description of [PHPLIB-252](https://jira.mongodb.org/browse/PHPLIB-252), which we can revisit later on. The idea there is that if we're seeking forward, advancing chunks withing the same cursor batch is free (the driver already has the data client-side) and a `getMore` still preferable to a new query (the server already has the data and we avoid the overhead of a new query and `killCursors` command for discarding the old cursor). However, if we're dealing with a very large file and want to seek to the end, it may very well be cheaper to query anew instead of walk the cursor with potentially many `getMore` commands.